source/codegen/service_helpers.py

```diff
@@ -419,6 +430,26 @@ def should_generate_mappings(enum_name: str) -> bool:
     function_enums = common_helpers.get_function_enums(functions)
     return [e for e in function_enums if should_generate_mappings(e)]
 
+def get_enums_to_map_type(enums: dict) ->List[str]:
+    """Get a list of the enums used by functions, for which mappings should be generated."""
+    list_of_enums: List[str] = []
+    for enum_name in enums.keys():
+        if "generate-mapping-type" in enums[enum_name]:
+            list_of_enums.append(enum_name)
+    return list_of_enums
+
+def get_type_from_enum(enums: dict) -> str:
```

Comment:
`generate_type_tag_enum_for_<your_name_for_this_feature>`. This is being used to select the field of a union. This is similar to "tagged" or "discriminated" unions, though in this case the "tag" is not part of the union.